### PR TITLE
Add qr code pairing to parent dashboard

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-14174
affects: Parent
release note: none

test plan
---------------
1. login to parent
2. click the add student button on dashboard
3. make sure modal menu is shown and that is shows 2 options, test both
4. disable flag `parentQRCodePairing` and make sure old behavior is still there where just the pairing alert comes up when you click add student

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/ixv5kik) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fixv5kik)
teacher: [Link](https://rebrand.ly/metds60) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fmetds60)
parent: [Link](https://rebrand.ly/mjnt5b5) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fmjnt5b5)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-14174](https://instructure.atlassian.net/browse/MBL-14174)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Parent/Parent/Dashboard/DashboardViewController.swift:308:1: Line Length Violation: Line should be 200 characters or less: currently 203 characters (line_length)
    Linting 'IBDesignables.swift' (46/88)
    Linting 'CourseDetailsViewController.swift' (47/88)
</details>
<details><summary>:x: XCTest failed: CoreTests/PlannerListViewControllerTests/testLayout</summary>


    Assertion Failure: PlannerListViewControllerTests.swift:89: XCTAssertEqual failed: ("nil") is not equal to ("Optional("note")")
    Assertion Failure: PlannerListViewControllerTests.swift:91: XCTAssertEqual failed: ("nil") is not equal to ("Optional(true)")
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Parent/Parent/Dashboard/SelectAddStudentMethodViewController.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        guard let row = AddObserveeMethod(rawValue: indexPath.row) else { fatalError("Invalid row") }
+        cell.textLabel?.text = row.description
+        cell.textLabel?.font = .scaledNamedFont(.medium16)
+        cell.textLabel?.textColor = .named(.textDarkest)
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        guard let method = AddObserveeMethod(rawValue: indexPath.row) else { fatalError("Invalid row") }
```
Comment (armstrongnate) [inline Parent/Parent/Dashboard/SelectAddStudentMethodViewController.swift:86]: I think we could return here instead of crashing.

**Comment by bluwave** [comment]
sounds good @armstrongnate

**Pull request merged**